Bug 68183: SXSSFWorkbook should dispose of temporary files when close() is called

InfiniteLoop90 opened the pull request:
This way consumers can use the workbook in a try-with-resources statement and the temp files will always be cleaned up without having to remember to explicitly call SXSSFWorkbook#dispose().

InfiniteLoop90 committed:
Bug 68183: SXSSFWorkbook should dispose of temporary files when close() is called

This way consumers can use the workbook in a try-with-resources statement and
the temp files will always be cleaned up without having to remember to explicitly
call SXSSFWorkbook#dispose().

InfiniteLoop90 commented:
Let me know if there’s anything else needed from me. Thanks!

centic9 commented:
This seems to break some test-cases, can you try to adjust them?

pjfanning commented:
* we can't change this behaviour unless the release is a major release
    * https://github.com/apache/poi/commit/eebb3717e04341aa041be14f8322ac4dd835c286 and related changes would also probably mean that the next release can't just be patch release (5.2.6) 
* `dispose` should be deprecated

InfiniteLoop90 commented:
Thanks for the feedback.

I made the following changes:

* Changed the `TestSXSSFWorkbookWithCustomZipEntrySource#validateTempFilesAreEncrypted` test so that it now passes.
* Added a new test to assert that temp files are disposed of when closing the SXSSFWorkbook without needing to call `SXSSFWorkbook#dispose`.
* Deprecated `SXSSFWorkbook#dispose`.

Ready for re-review when ready. Thanks again.

pjfanning commented:
thanks - merged
